app/data-generation/check-dax-prn.js

```diff
@@ -0,0 +1,27 @@
+const db = require('../data')
+const { PAYMENT_ACKNOWLEDGED_STATUS, PAYMENT_SETTLED_STATUS } = require('../constants/statuses')
+const { getDataFilter } = require('../get-data-filter')
+const { getStatus } = require('./get-status')
+
+const checkDAXPRN = async (event, transaction) => {
+  const status = getStatus(event)
+  if ([PAYMENT_ACKNOWLEDGED_STATUS, PAYMENT_SETTLED_STATUS].includes(status)) {
+    return event.data.paymentRequestNumber
+  }
+  for (let i = event.data.paymentRequestNumber; i > 0; i--) {
+    const where = getDataFilter(event.data)
```

Comment:
This loop could be more efficient by using for (const previousRequest of previousRequests) and loading the database query once instead of each time you loop over. By using FindAll instead of FindOne you cut down on query numbers as long as the dataset isn't prohibitively large